packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm

```diff
@@ -148,7 +148,9 @@ - (void)didMoveToWindow
 {
   [super didMoveToWindow];
   if (self.window) {
-    [self _attach];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [self _attach];
+    });
   } else {
     [self _detach];
```

Comment:
shouldn't we add the same in the detach? also, how come that didMoveToWindow is called in a queue that is not the main queue? These methods are called by UIKit, React Native is not calling them explicitly...

Reply:
I didn't notice any problems with detach behaviour therefore I've left the code intact. 

I believe this is a timing issue rather. The method is executed on UI thread as expected. See [below](https://github.com/facebook/react-native/pull/48580#issuecomment-2582382187)